File: 3. Semestar/Napredno programiranje i programski jezici/Vezbe/Vezbe 3/1/valjak.cpp
#include "valjak.hpp"

Valjak::Valjak(double r, double b): baza(r), omotac(2*r*3.14, b){}

double Valjak::getR(){
    return baza.getR();
}

double Valjak::getH(){
    return omotac.getB();
}

double Valjak::getV(){
    return omotac.getB() * baza.getR();
}

double Valjak::getP(){
    return 2 * baza.getP() + omotac.getP();
}

File: 3. Semestar/Napredno programiranje i programski jezici/Vezbe - Nasledjeno/vezbe9/Njihova resenja/3/Valjak.java
package z3;

public class Valjak {
	private Krug b;
	private Pravougaonik m;
	
	public Valjak(double r, double h) {
		this.b = new Krug(r);
		this.m = new Pravougaonik(b.getO(), h);
	}
	
	public Valjak(Valjak v) {
		this.b = new Krug(v.b);
		this.m = new Pravougaonik(v.m);
	
	}
	public double getR() {
		return b.getR();
	}
	
	public double getH() {
		return m.getB();
	}
	
	public double getP() {
		return 2 * b.getP() + m.getP();
	}
	
	public double getV() {
		return b.getP() * getH();
	}

}
